src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -448,13 +448,35 @@ protected function forward(Request $request, $catch = false, Response $entry = n
         // always a "master" request (as the real master request can be in cache)
         $response = SubRequestHandler::handle($this->kernel, $request, HttpKernelInterface::MASTER_REQUEST, $catch);
 
-        // we don't implement the stale-if-error on Requests, which is nonetheless part of the RFC
-        if (null !== $entry && \in_array($response->getStatusCode(), [500, 502, 503, 504])) {
+        /*
+         * Support stale-if-error given on Responses or as a config option.
+         * RFC 7234 summarizes in Section 4.2.4 (but also mentions with the individual
+         * Cache-Control directives) that
+         *
+         *      A cache MUST NOT generate a stale response if it is prohibited by an
+         *      explicit in-protocol directive (e.g., by a "no-store" or "no-cache"
+         *      cache directive, a "must-revalidate" cache-response-directive, or an
+         *      applicable "s-maxage" or "proxy-revalidate" cache-response-directive;
+         *      see Section 5.2.2).
+         *
+         * https://tools.ietf.org/html/rfc7234#section-4.2.4
+         */
+        if (null !== $entry
+            && \in_array($response->getStatusCode(), [500, 502, 503, 504])
+            && !$entry->headers->hasCacheControlDirective('no-cache')
+            && !$entry->mustRevalidate()
+            && !$entry->headers->hasCacheControlDirective('s-maxage')
```

Comment:
I don't get it. If you bypass it when s-maxage is set, that limits the usefulness of the feature drastically.

Reply:
Same for me.

Also see the last sentence in https://github.com/httpwg/http-core/issues/264#issuecomment-557979200 and verify at https://tools.ietf.org/html/rfc7234#section-5.2.2.9.

Reply:
Let's revert that part.

Reply:
You mean only for `smax-age`, right? Fine with that.

I'll also see if i can get clarifying comments from the RFC editors. Guess the bottom line is that `max-age + public != s-maxage` as I always thought.